modules/dns-issuer/provider.tf

```diff
@@ -0,0 +1,19 @@
+data "digitalocean_kubernetes_cluster" "cluster" {
+  name = var.cluster_name
+}
+
+provider "kubernetes" {
```

Comment:
I think will be good to not have provider blocks digitalocean/kubernetes/kubectl in this module, and instead use them in outside module/code,
this type construction prevents to use this module in conditional and loop operations,
here we can just have requirement in terraform.required_providers{} block,

then we can in k8s-cluster module just use this module by having it under enable/disable switcher condition like we have it for cert-manager, nginx-ingress and etc.